arch/xtensa/include/irq.h

```diff
@@ -155,12 +168,6 @@ struct xcptcontext
 
   uint32_t *regs;
 
-#if XCHAL_CP_NUM > 0
-  /* Co-processor save area */
-
-  struct xtensa_cpstate_s cpstate;
```

Comment:
It looks like this is fixed for only one owner of the coprocessors and saving/restoring in any context switch.  `struct xtensa_cpstate_s` also contained other members to help managing different owners of coprocessors.

Reply:
The reason I move the `struct xtensa_cpstate_s cpstate` is:  lazy coprocessors registers context switching is not support now, and the `cpenable` and `cpstored` is always enable and same in every task.